sdk/python/kfp/compiler/compiler.py

```diff
@@ -793,7 +792,8 @@ def _create_workflow(self,
       raise ValueError('Either specify pipeline params in the pipeline function, or in "params_list", but not both.')
 
     args_list = []
-    for arg_name in argspec.args:
+    signature = inspect.signature(pipeline_func)
```

Comment:
nit: adding a test case? this seems like a very subtle change

Reply:
I'm not sure we need to explicitly support `@wraps`. It's just that the old functions like `getfullargspec` do not understand it. Those functions were planned to be deprecated, but then were kept.

Reply:
got it, SGTM